docs/src/contributing/testnet.md

```diff
@@ -0,0 +1,237 @@
+  * Source your shell environment to be sure the new `SOLANA_METRICS_CONFIG` is loaded
+* By default, metrics are only logged by agave if `RUST_LOG` is set to `info` or higher. You can set this in your shell environment as well.
+  ```bash
+  RUST_LOG="info,solana_runtime=debug"
+  ```
+
+### To validate that your environment is set up 100% correctly
+
+```bash
+  cd ./scripts/
+  source  ./configure-metrics.sh
+    INFLUX_HOST=https://metrics.solana.com:9999
+    INFLUX_DATABASE=testnet-dev-solana
+    INFLUX_USERNAME=solana
+    INFLUX_PASSWORD=********
+  ./metrics-write-datapoint.sh "testnet-deploy net-create-begin=1"
+
+  ```
+  * All commands should complete with no errors, this indicates your influxDB config is usable
+  * Ensure that `RUST_LOG` is set to `info` or `debug`
+
+## Quick Start
+
+NOTE: This example uses GCE.  If you are using AWS EC2, replace `./gce.sh` with
+`./ec2.sh` in the commands.
+
+```bash
+cd net/
+
+# Recreate a metrics database for the testnet and validate credentials (if you are using metrics)
+./init-metrics.sh -e
+
+# Create a GCE testnet with 4 additional validator nodes (beyond the bootstrap node) and 1 client (billing starts here)
+./gce.sh create -n 4 -c 1
+
+# Deploy the network from the local workspace and start processes on all nodes including bench-tps on the client node
+RUST_LOG=info ./net.sh start
+
+# Show a help to ssh into any testnet node to access logs/etc
+./ssh.sh
+
+# Stop running processes on all nodes
+./net.sh stop
+
+# Dispose of the network (billing stops here)
+./gce.sh delete
+```
+
+## Full guide
+* If you expect metrics to work, make sure you have configured them before proceeding
+* Go to `./net/` directory in agave repo
+* `./gce.sh` command controls creation and destruction of the nodes in the test net. It does not actually run any software.
+  * `./gce.sh create \-n 4 \-c 2` creates cluster with 4 validators and 1 node for load generation, this is minimal viable setup for all solana features to work
+  * `./gce.sh info`  lists active test cluster nodes, this allows you to get their IP addresses for SSH access and/or debugging
+  * `./gce.sh delete`  destroys the nodes (save the electricity and $$$ - destroy your test nets the moment you no longer need them).
+  * On GCE, if you do not delete nodes, they will self-destruct in 8 hours anyway, you can configure self-destruct timer by supplying `--self-destruct-hours=N` argument to `gce.sh`
+  * On other cloud platforms the testnet will not self-destruct!
+* `./net.sh` controls the payload on the testnet nodes, i.e. bootstrapping, the validators and bench-tps. In principle, you can run everything by hand, but `./net.sh` makes it easier.
+  * `./net.sh start` to actually run the test network.
+    * This will actually upload your current sources to the bootstrap host, build them there and upload the result to all the nodes
+    * The script will take 5-10 of minutes to run, in the end it should print something like
+     ```
```

Comment:
this sort of thing is obviously subject to change and thus fragile. I know you hedged by saying "something like", but I wonder if we can use something else to indicate success.. like maybe some RPC query to ensure things are up

Reply:
Left only the "Deployment successful" part as that is hardwired into the script